cartoframes/data/dataset.py

```diff
@@ -688,7 +683,7 @@ def compute_geom_type(self):
         if self._state == Dataset.STATE_REMOTE:
             return self._get_remote_geom_type(get_query(self))
         elif self._state == Dataset.STATE_LOCAL:
-            return self._get_local_geom_type(get_geodataframe(self))
+            return self._get_local_geom_type()
```

Comment:
We should pass here always a geodataframe. Now the function internally uses `self.dataframe.geometry`, but since `self.dataframe` can be a GeoDataFrame or a DataFrame we can not make sure that it contains a `geometry` attribute.

Reply:
Since `compute_geom_type` is a method only used by `Source` and the source Source constructor is creating the geodataframe if possible.

Reply:
We can check if geometry exists, but then, it will fail in another place

Reply:
Yep, this is the use case now, but my concern is about the fragility if we use this method in other flow in the future. However, since it works for this case, I would move it forward and maybe revisit it in the future if required

Reply:
:+1: And it is also out of the scope of the issue. However, the next PR using the strategy pattern will help to have a better Dataset API